reboot_pending/reboot_pending.dsc.resource.json

```diff
@@ -1,16 +1,27 @@
 {
-    "$schema": "https://raw.githubusercontent.com/PowerShell/DSC/main/schemas/2024/04/bundled/resource/manifest.json",
-    "description": "Returns info about pending reboot.",
-    "type": "Microsoft.Windows/RebootPending",
-    "version": "0.1.0",
-    "get": {
-      "executable": "powershell",
-      "args": [
-        "-NoLogo",
-        "-NonInteractive",
-        "-NoProfile",
-        "-Command",
-        "reboot_pending.resource.ps1"
-      ]
+  "$schema": "https://raw.githubusercontent.com/PowerShell/DSC/main/schemas/2024/04/bundled/resource/manifest.json",
+  "description": "Returns info about pending reboot.",
+  "type": "Microsoft.Windows/RebootPending",
+  "version": "0.1.0",
+  "get": {
+    "executable": "powershell",
+    "args": [
+      "-NoLogo",
+      "-NonInteractive",
+      "-NoProfile",
+      "-Command",
+      "reboot_pending.resource.ps1"
+    ]
+  },
+  "schema": {
+    "embedded": {
+      "$schema": "https://json-schema.org/draft/2020-12/schema",
+      "type": "null",
```

Comment:
I think type should be `object`

Reply:
With `object` I get 
`ERROR Error: Schema: null is not of type "object"`

Reply:
The existing schema we apply to embedded commands _requires_ that `type` always be object ([docs][01], [schema source][02]).

[01]: https://learn.microsoft.com/powershell/dsc/reference/schemas/resource/manifest/schema/embedded?view=dsc-3.0&preserveView=true#type
[02]: https://github.com/PowerShell/DSC/blob/6483333b9ef689e795761d17428d2ff3ff2768a3/schemas/src/resource/manifest.schema.yaml#L114-L125

We could change this to allow users to specify the valid options as `object`, `null`, or `[object, null]` but I think it probably easier on resource authors to hand over an empty object when `properties` doesn't define any key-value pairs.

Reply:
Actually, the type can't be just `null`, it needs to be either `object` or `[object, null]` - otherwise defining the properties keyword in the schema means nothing, and the schema effectively can't be used to validate the output.

Reply:
@michaeltlombardi so what should be the `type` in this bug fix?

Reply:
It should be `[object, null]`

Reply:
> It should be `[object, null]`

This fails with:
`Error: Schema: "[object, null]" is not valid under any of the schemas listed in the 'anyOf' keyword`

Reply:
We may have to also update the embedded schema definition for this to work. That would mean editing this line:

https://github.com/PowerShell/DSC/blob/6483333b9ef689e795761d17428d2ff3ff2768a3/schemas/src/resource/manifest.schema.yaml#L125

to:

```yaml
enum:
- object
- [object, 'null']
```

I _think_. I would need to do some quick local testing.

I'm still not confident we _should_ support passing `null` to the instance instead of an empty object - right now, an empty object would still be valid, because there are no required properties.

A schema with no required properties can receive an empty object and still be valid.

I can spin up a local change of the schema and submit a PR updating it if we definitely want to support both `object` and `[object, 'null']` for the valid types.

Reply:
I'm ok with an empty object vs null as it implies to me reading a config something deliberate vs accidental

Reply:
ok, so what should the schema look like in this case?

Reply:
The `type` should be `object` - you mentioned earlier that you got the error `ERROR Error: Schema: null is not of type "object"` - was that from a configuration command or direct resource invocation?

Reply:
config command:
`dsc config get --path C:\temp\reboot_pending.yml`

Reply:
@anmenaga can you add an example yaml for rebootPending?

Reply:
Added  example yaml and a test that uses it.